### PR TITLE
relates to #1730: added document delete resource test

### PR DESCRIPTION
**Pull request by EricBorczuk**
**What this PR does**:

**Which issue(s) this PR fixes**:
Relates #1730 

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+  public static final String DEFAULT_PAYLOAD =
+      "{\"test\": \"document\", \"this\": [\"is\", 1, true]}";
+
+  @Inject NamespaceManager namespaceManager;
+
+  @Inject TableManager tableManager;
+
+  @Inject WriteDocumentsService writeDocumentsService;
+
+  @Inject ObjectMapper objectMapper;
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:47]: do you actually need this?

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+            tableManager.getValidCollectionTable(DEFAULT_NAMESPACE, DEFAULT_COLLECTION),
+            DEFAULT_NAMESPACE,
+            DEFAULT_COLLECTION,
+            DEFAULT_DOCUMENT_ID,
+            objectMapper.readTree(DEFAULT_PAYLOAD),
+            null,
+            ExecutionContext.NOOP_CONTEXT)
+        .await()
+        .atMost(Duration.ofSeconds(10));
+  }
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:78]: 1. let's write through the api as well?

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+  @Nested
+  class DeleteDocument {
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:81]: general stuff:

* let's have `2xx` cases first, then `4xx`, that way we can write code in some meaningful way and we know where is what added
* let's rename tests to exclude `test` prefix, `testDelete` -> `delete`

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@QuarkusTest
+@TestProfile(IntegrationTestProfile.class)
+@ActivateRequestContext
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class DocumentDeleteResourceIntegrationTest {
+
+  public static final String BASE_PATH = "/v2/namespaces/{namespace}/collections/{collection}";
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:34]: you can also include `"/v2/namespaces/{namespace}/collections/{collection}/{document-id}"` here so you don't have to repeat it all the time

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+          .then()
+          .statusCode(404);
+
+      // When a delete occurs on an unknown document, it still returns 204 No Content
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .delete(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(204);
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:137]: this is enough for this test, no need to get before or after

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+          .when()
+          .get(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
+          .then()
+          .statusCode(404);
+
+      // When a delete occurs on an unknown document, it still returns 204 No Content
+      given()
+          .header(Constants.AUTHENTICATION_TOKEN_HEADER_NAME, "")
+          .when()
+          .delete(BASE_PATH + "/no-id", DEFAULT_NAMESPACE, DEFAULT_COLLECTION)
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:135]: `id` as last parameter?

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java (hunk 3 above)
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:81]: missing cases for:

* table does not exist
* keyspace does not exist

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+          .when()
+          .delete(
+              BASE_PATH + "/{document-id}/test",
+              DEFAULT_NAMESPACE,
+              DEFAULT_COLLECTION,
+              DEFAULT_DOCUMENT_ID)
+          .then()
+          .statusCode(204);
+
+      given()
```
Comment (ivansenic) [inline sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:190]: let's also get a whole document and ensure there is no `test`:

`.body("test", nullValue());`

**Review by ivansenic — APPROVED** [review]
* Why is this a draft, it seems done?
* This must re re-based on master before merged (depends on  #1931)

:rocket:

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java (hunk 7 above)
Comment (ivansenic) [reply sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:190]: same for the array case below

File: sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java (hunk 2 above)
Comment (EricBorczuk) [reply sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:78]: Would that be outside of the scope of "testing deletion" though?

Comment (ivansenic) [reply sgv2-docsapi/src/test/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResourceIntegrationTest.java:78]: does not matter, when we test deletion of collection, we do write it via the API.. We need to confirm that and not that we are able to delete the document that's being written through the injected service..

**Comment by ivansenic** [comment]
@EricBorczuk Ci reports the failing tests, I will fix this and merge:

```
Error:  Failures: 
Error:    DocumentDeleteResourceIntegrationTest$DeleteDocument.deleteNotFound:117 1 expectation failed.
Expected status code <404> but was <500>.
Error:    DocumentDeleteResourceIntegrationTest$DeleteDocument.happyPath:89 1 expectation failed.
Expected status code <404> but was <500>.
Error:    DocumentDeleteResourceIntegrationTest$DeleteDocumentPath.testDeletePath:163 1 expectation failed.
Expected status code <404> but was <500>.
```

**Pull request merged**